compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -985,22 +985,21 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         let ns = source.namespace();
         let is_expected = &|res| source.is_expected(res);
 
-        let path_sep = |err: &mut Diagnostic, expr: &Expr| match expr.kind {
-            ExprKind::Field(_, ident) => {
+        let path_sep = |err: &mut Diagnostic, expr: &Expr| match &expr.kind {
+            ExprKind::Field(base, ident) => {
                 err.span_suggestion(
-                    expr.span,
+                    base.span.source_callsite().between(ident.span),
```

Comment:
I use `.source_callsite()` to avoid creating an extra incorrect / irrelevant highlight inside of the macro definition.
Without it, the diagnostic would look like this:

```
error[E0423]: expected value, found trait `std::convert::Into`
  --> q.rs:2:13
   |
2  |       () => { ::std::convert::Into::<()> }
   |  _____________^^^^^^^^^^^^^^^^^^^^^^^^^^-
3  | | }
4  | |
5  | | fn main() {
...  |
15 | |
16 | |     let you = make!().into(());
   | |_______________-------- help: use the path separator to refer to an item: `::`
   |                 |
   |                 in this macro invocation
   |
   = note: this error originates in the macro `make` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Notice the small `-` below whitespace inside of the macro definition.

Reply:
Can you use `Span::find_ancestor_inside` with `expr.span` instead?